opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/fortran/FortranAnalyzerFactory.java

```diff
@@ -49,7 +49,9 @@ public class FortranAnalyzerFactory extends FileAnalyzerFactory {
         "F95",
         "F03",
         "F08",
-        "F15"};
+        "F15",
+        "F77",
+        "I"};
```

Comment:
I wonder whether the `.i` suffix is perhaps too generic to overlap with other languages or file types ?

Reply:
.i is more like .h for Fortran 77. It contains the declarations.

Reply:
This suffix is already used by C: https://github.com/oracle/opengrok/blob/f92514eb963f886272c07dbd384a978dbf7b6a76/opengrok-indexer/src/main/java/org/opengrok/indexer/analysis/c/CAnalyzerFactory.java#L36 and according to the current rules certain suffix can be used in only one analyzer. The `.i` files appear to be those C preprocessed: https://stackoverflow.com/a/56482777/11582827

Reply:
This can be resolved during runtime by using the `-A` option of the indexer, i.e. `-A .i:FortranAnalyzer`. This will override the C analyzer with Fortran analyzer.

Reply:
It is the tests that fail for suffix I

Reply:
yes, the `AnalyzerGuru` contains self-consistency checks that prevent adding suffix that is already present in an analyzer.